agent/agent.go

```diff
@@ -297,3 +305,29 @@ func (a *Agent) killProcs(tc *taskContext) {
 	}
 	grip.Infof("processes cleaned up for task %s", tc.task.ID)
 }
+
+func (a *Agent) runSetupScript(ctx context.Context) (string, error) {
```

Comment:
in what context do we need more than an error? 

should this function log more

Reply:
If the provisioning script hits an error, we want to return it.

Reply:
yeah, but we just annotate the error with the out message (which isn't not weird)

Reply:
If we do that though, we can't test that `runSetupScript` actually ran the script.

Reply:
can we return, say, an exit code as an integer, log the output, and annotate the error as needed, rather than needing to ready any information into the string value?

Reply:
If we return an integer, we can't distinguish between no-oping because the script isn't there and running the script successfully.